src/targets/mobile/lib/withPersistentState.js

```diff
@@ -0,0 +1,20 @@
+import localforage from 'localforage'
+
+// The withPersistentState HOC automatically saves the wrappped components state and restores it when the component is mounted again.
+// Since the saved state will not be available immediately when the wrapped component is mounted, a `componentStateRestored` hook will be called once the state is restored
+export const withPersistentState = (WrappedComponent, persistenceKey) => class extends WrappedComponent {
+  async componentWillMount () {
+    if (super.componentWillMount) super.componentWillMount()
+    const persistedState = await localforage.getItem(persistenceKey)
+    this.setState(prevState => (persistedState || prevState), () => {
+      if (super.componentStateRestored) super.componentStateRestored()
+    })
```

Comment:
Ok, I saw this after my interrogation:

> Since the saved state will not be available immediately when the wrapped component is mounted, a `componentStateRestored` hook will be called once the state is restored

If I understand well, you suggest to implement such a function to let the component do something once the state is set.
It appears that React already have such a function in the component's lifecycle called `componentDidUpdate` (which is called after the `render()`) or `componentWillUpdate` (which is called before the `render()`).
Do you thing this `componentStateRestored` is mandatory and in which usecase?

More details in the documentation: https://facebook.github.io/react/docs/react-component.html#updating